components/popover/index.js

```diff
@@ -133,17 +133,27 @@ class Popover extends Component {
 	}
 
 	setOffset() {
+		const { range } = this.props;
 		const { anchor, popover } = this.nodes;
 		if ( ! anchor || ! anchor.parentNode ) {
 			return;
 		}
 
-		const rect = anchor.parentNode.getBoundingClientRect();
 		const [ yAxis, xAxis ] = this.getPositions();
 		const isTop = 'top' === yAxis;
 		const isLeft = 'left' === xAxis;
 		const isRight = 'right' === xAxis;
-
+		let rect = anchor.parentNode.getBoundingClientRect();
+		if ( range ) {
+			const rects = range.getClientRects();
+			if ( isLeft ) {
+				rect = rects[ 0 ];
+			} else if ( isRight ) {
+				rect = rects[ rects.length - 1 ];
```

Comment:
Reading through the documentation for [`Element#getClientRects`](https://developer.mozilla.org/en-US/docs/Web/API/Element/getClientRects), it's not clear to me what we're achieving by referencing the start/end of the returned collection?

Reply:
For the case of auto-completion it will do nothing because all the text will be on one line however if this was used on something that could be line wrapped then positioning over the first rect for left alignment and the last rect for right alignment seems to be sensible to me.